dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -106,7 +106,7 @@ public class GitHubSourceCodeRepo extends SourceCodeRepoInterface {
 
     public static final String OUT_OF_GIT_HUB_RATE_LIMIT = "Out of GitHub rate limit";
     public static final String GITHUB_ABUSE_LIMIT_REACHED = "GitHub abuse limit reached";
-    public static final int GITHUB_DEFAULT_MAX_CACHE_AGE_SECONDS = 60;
+    public static final int GITHUB_MAX_CACHE_AGE_SECONDS = 30; // GitHub's default max-cache age is 60 seconds
```

Comment:
I know Steve suggested decreasing it and I didn't comment on it, but I don't see the need to change the cache to be less than GitHub's own. But if tests work, it doesn't hurt, so you don't have to change.

Reply:
To clarify, I was concerned, not knowing the exact details of how GitHub sets the relevant HTTP headers, and how the `OkHttp` class actually uses the `MAX_CACHE_AGE`, that we could possibly be introducing another change propagation delay, on top those already built in to GitHub's systems.  For example, if the repo info update takes 20 seconds between when we make the change on the site and when the change first appears in the API response, and we add another caching delay of 60 seconds, the total is 80 seconds.  That might be pretty confusing to us and/or our users in certain situations (imagine, for example, you're testing the "auto topic" harvesting feature, and the change isn't showing up...)

Reply:
> For example, if the repo info update takes 20 seconds between when we make the change on the site and when the change first appears in the API response, and we add another caching delay of 60 seconds, the total is 80 seconds

If the repo info update currently takes 20 seconds for changes on GitHub to be reflected on Dockstore, then that means that our current cache max-age is 20 seconds, but I don't believe this is true based on my topics testing on prod which took about 50 seconds for the change to show up on Dockstore (timing might be slightly conservative because time is wasted clicking between tabs to change the topic, etc).

If `MAX_CACHE_AGE` is set to 60 seconds, then this would be the amount of time it would take for a change on GitHub to be reflected on Dockstore. I don't think there is an additional delay, unless I'm interpreting things wrong. The `MAX_CACHE_AGE` that the `OkHttpClient` uses is added as a `max-age` cache control header when requests are made. If we set it to 60 seconds, then the cached response is used if it's age doesn't exceed 60 seconds. 

Also, to clarify, I found GitHub's max-cache age default of 60 seconds by looking at the response header from `curl -I https://api.github.com/users/octocat/orgs`. This default isn't built into the kohsuke GitHub library if that was adding any confusion.

Reply:
Cool, as the implementer, you have the best visibility into this issue, so let's go with whatever setting you recommend...